tests/rgen/Microsoft.Macios.Generator.Tests/Classes/Data/ExpectedMethodsTests.cs

```diff
@@ -317,17 +321,42 @@ public virtual unsafe partial void CompleteRequest (global::Foundation.NSExtensi
 		return ret;
 	}
 
+	[SupportedOSPlatform ("macos")]
+	[SupportedOSPlatform ("ios")]
+	[SupportedOSPlatform ("tvos")]
+	[SupportedOSPlatform ("maccatalyst13.1")]
+	[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+	public static unsafe partial global::Foundation.NSData GetBookmarkData (global::Foundation.NSUrl bookmarkFileUrl, string subdomain, out global::Foundation.NSError? error)
+	{
+		if (subdomain is null)
+			global::ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (subdomain));
+		var bookmarkFileUrl__handle__ = bookmarkFileUrl!.GetNonNullHandle (nameof (bookmarkFileUrl));
+		var nssubdomain = global::CoreFoundation.CFString.CreateNative (subdomain);
```

Comment:
Another idea (which might be a good thing anyway) would be to use TransientCFString:

```cs
using var nssubdomain = global::ObjCRuntime.TransientCFString (subdomain);
```

which has a couple of benefits:

* No need to free it manually, it'll happen automatically.
* Won't leak if there are any exceptions at any point.

Downsides:

* It won't work in third-party bindings, because `TransientCFString` is internal.

Reply:
Good point, I'll add that as an issue. I can check what .dll is being built and generate code based on that.